src/TypedStateMachines/Events/StateChange.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace TypedStateMachines\Events;
+
+use TypedStateMachines\IEvent;
+use TypedStateMachines\IStateMachine;
+use TypedStateMachines\Transition;
+
+class StateChange implements IEvent
+{
+
+    /**
+     * @var IStateMachine
+     */
+    public $stateMachine;
+
+    /**
+     * @var Transition
+     */
+    public $transition;
+
+    /**
+     * Constructor for the StateChange event.
+     *
+     * @param IStateMachine $stateMachine
+     * @param Transition $transition
+     */
+    public function __construct(IStateMachine $stateMachine, Transition $transition)
+    {
+        $this->stateMachine = $stateMachine;
+        $this->transition   = $transition;
+    }
+
+    /**
+     * Fetch the name of the Event.
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        return 'state_change';
```

Comment:
Nitpicy but constants?

Reply:
I'm not sure where to place that constant? 
Given that it's only used here. 
I don't think it should be defined in the StateMachine.